static-assets/components/cstudio-forms/controls/input.js

```diff
@@ -169,7 +175,9 @@ YAHOO.extend(CStudioForms.Controls.Input, CStudioForms.CStudioFormField, {
     this.inputEl = inputEl;
     YAHOO.util.Dom.addClass(inputEl, 'datum');
     YAHOO.util.Dom.addClass(inputEl, 'cstudio-form-control-input');
-    inputEl.value = (this.value = '_not-set') ? config.defaultValue : this.value;
+
+    const valueToSet = this.escapeContent ? CStudioForms.Util.unEscapeXml(this.value) : this.value;
+    inputEl.value = (this.value = '_not-set') ? config.defaultValue : valueToSet;
```

Comment:
This is an odd statement. Looks like it will always be '_not-set' and will always use config.defaultValue.

`(this.value = '_not-set')` always sets the value and returns that string, that value is always truthy so condition is always true.

Perhaps this was meant to check equality of this.value instead of assign it? Please review.

Reply:
yes, I checked the code and seems like the intention was to check equality instead of assigning value